data/debian/sonic-host-services-data.process-reboot-cause.service

```diff
@@ -1,7 +1,7 @@
 [Unit]
 Description=Retrieve the reboot cause from the history files and save them to StateDB
-Requires=database.service determine-reboot-cause.service
 After=database.service determine-reboot-cause.service
+Upholds=database.service determine-reboot-cause.service
```

Comment:
If I'm understanding `Upholds=` correctly, this would mean that if `database.service` fails to start up for whatever reason, then this service can still start, since there's no longer a hard dependency on the database container. However, it looks like process-reboot-cause writes to state DB, which would then fail, and then cause this service to fail.

Reply:
Yes, that is expected, If the database service did not start correctly, the process-reboot-cause will not start correctly which will failing in read db as expected, so it will not persist any cause into db until next reboot and database service started correctly.